features/cellular/framework/AT/AT_CellularStack.cpp

```diff
@@ -158,23 +157,35 @@ nsapi_error_t AT_CellularStack::socket_stack_init()
 
 nsapi_error_t AT_CellularStack::socket_open(nsapi_socket_t *handle, nsapi_protocol_t proto)
 {
-    if (!is_protocol_supported(proto) || !handle) {
+    if (!handle) {
         return NSAPI_ERROR_UNSUPPORTED;
     }
 
-    int max_socket_count = get_max_socket_count();
+    if (proto == NSAPI_UDP) {
+        if (!_device.get_property(AT_CellularDevice::PROPERTY_IP_UDP)) {
```

Comment:
Cosmetic, but maybe these could be combined into one if -sentence, like:
`    if (proto == NSAPI_UDP && !_device.get_property(AT_CellularDevice::PROPERTY_IP_UDP)) {`